### PR TITLE
[WIP] for examples

### PR DESCRIPTION
**Pull request by gavin1332**

### CODE REVIEW

File: dygraph/mnist/train.py [hunk 1]
```diff
@@ -185,11 +191,18 @@ def train_mnist(args):
...
+            role = role_maker.PaddleCloudRoleMaker(is_collective=True)
+            fleet.init(role)
         mnist = MNIST()
         adam = AdamOptimizer(learning_rate=0.001, parameter_list=mnist.parameters())
+
         if args.use_data_parallel:
-            mnist = fluid.dygraph.parallel.DataParallel(mnist, strategy)
+            #mnist = fluid.dygraph.parallel.DataParallel(mnist, strategy)
+            dist_strategy = fleet.DistributedStrategy()
+            engine = fleet.distributed_optimizer(adam, dist_strategy, model=mnist)
```
Comment (guru4elephant) [inline dygraph/mnist/train.py]: should consider seriously.

Comment (guru4elephant) [reply dygraph/mnist/train.py]: how about you just add a function for fleet to set model.

Comment (guru4elephant) [reply dygraph/mnist/train.py]:
``` python
fleet.init(is_collective=True)
mnist = fleet.get_distributed_model(mnist) # in dygraph mode, set_model must be called
dist_strategy = fleet.DistributedStrategy()
adam = fleet.distributed_optimizer(adam, dist_strategy) # adam has mnist.


```

Comment (gavin1332) [reply dygraph/mnist/train.py]: `fleet.get_distributed_model(mnist)` should call after distributed_optimizer so as to apply `dist_strategy`

**Pull request closed without merging**